### PR TITLE
Added support for OpenDistro in integration tests

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Added support to run integration tests on `OpenDistro` and even on `Elasticsearch`.
Command line sample is:
```bash
./build.sh integrate opendistro-latest writable random:test_only_one --report
```
Used abstractions library update is able to run tests on `OpenDistro` versions `1.13.0-1.13.3` (`opendistro-latest` resolved as `1.13.3`) and `Elasticsearch` `7.10.2` (`elasticsearch-latest` resolved as `7.10.2`). Other versions are incompatible with OpenSearch, so are not supported by the test framework as well.

## Notes
* 3a9fcdf required to run integration tests.
* `latest` resolution for `OpenSearch` is not implemented yet.
* `OpenSearch` is shipped as tarball for Linux only, testing `OpenDistro` and `Elasticsearch` on Windows might fail, because it not supported [yet].

### CODE REVIEW

File: tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs [hunk 1]
```diff
@@ -63,8 +63,8 @@ public static ConnectionSettings CreateConnectionSettings<TConfig>(this IEphemer
 
 				var noCertValidation = current.ServerCertificateValidationCallback == null;
 
-				if (cluster.ClusterConfiguration.EnableSecurity && notAlreadyAuthenticated)
-					settings = settings.BasicAuthentication(ClusterAuthentication.Admin.Username, ClusterAuthentication.Admin.Password);
+				//if (cluster.ClusterConfiguration.EnableSecurity && notAlreadyAuthenticated)
```
Comment (ghost) [inline tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs]: Should this be removed?

File: tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs [hunk 2]
```diff
@@ -63,8 +63,8 @@ public static ConnectionSettings CreateConnectionSettings<TConfig>(this IEphemer
...
+				//	settings = settings.BasicAuthentication(ClusterAuthentication.Admin.Username, ClusterAuthentication.Admin.Password);
```
Comment (alexmeizer) [inline tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs]: remove commented code here

**Review by alexmeizer — CHANGES_REQUESTED** [review]
Additionally, get in the habit of adding new lines at end of files if you see it missing one and you're touching it. See: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log

**Review by alexmeizer — APPROVED** [review]

File: tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs (hunk 2 above)
Comment (Yury-Fridlyand) [reply tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs]: Thank you, fixed in 2982ff909.

File: tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs (hunk 1 above)
Comment (Yury-Fridlyand) [reply tests/Tests.Core/Extensions/EphemeralClusterExtensions.cs]: Thank you, fixed in 2982ff909.

**Review by ghost — APPROVED** [review]

**Review by MaxKsyunz — APPROVED** [review]

**Pull request merged**